src/awaitVercelDeployment.ts

```diff
@@ -1,38 +1,50 @@
 import * as core from '@actions/core';
 import fetch from '@adobe/node-fetch-retry';
+import { setTimeout } from 'timers/promises';
 import { VERCEL_BASE_API_ENDPOINT } from './config';
 import { VercelDeployment } from './types/VercelDeployment';
 
+interface Options {
+  /** Base url of the Vercel deployment to await for */
+  url: string;
+  /** Duration (in milliseconds) to wait for a terminal deployment status */
+  timeout: number;
+  /** Duration (in milliseconds) to wait in between polled Vercel API requests */
+  pollInterval: number;
+}
+
 /**
  * Awaits for the Vercel deployment to be in a "ready" state.
  *
- * @param baseUrl Base url of the Vercel deployment to await for.
- * @param timeout Duration (in seconds) until we'll await for.
- *  When the timeout is reached, the Promise is rejected (the action will fail).
+ * When the `timeout` is reached, the Promise is rejected (the action will fail)
  */
-const awaitVercelDeployment = (baseUrl: string, timeout: number): Promise<VercelDeployment> => {
+const awaitVercelDeployment = ({ url, timeout, pollInterval }: Options): Promise<VercelDeployment> => {
   return new Promise(async (resolve, reject) => {
     let deployment: VercelDeployment = {};
     const timeoutTime = new Date().getTime() + timeout;
 
     while (new Date().getTime() < timeoutTime) {
-      deployment = (await fetch(`${VERCEL_BASE_API_ENDPOINT}/v11/now/deployments/get?url=${baseUrl}`, {
+      const retryMaxDuration = timeoutTime - new Date().getTime(); // constrain retries by remaining timeout duration
+
+      core.debug(`Retrieving deployment (retryMaxDuration=${retryMaxDuration}ms)`);
+      deployment = await fetch(`${VERCEL_BASE_API_ENDPOINT}/v11/now/deployments/get?url=${url}`, {
         headers: {
           Authorization: `Bearer ${process.env.VERCEL_TOKEN}`,
         },
-        retryOptions: {
-          retryMaxDuration: timeout * 1000, // Convert seconds to milliseconds
-        },
-      })
-        .then((data) => data.json())
-        .catch((error: string) => reject(error))) as VercelDeployment;
```

Comment:
Given we were `await`ing for the promise, this didn't seem necessary. Errors thrown inside of executor function will [inherently reject the promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise#parameters).